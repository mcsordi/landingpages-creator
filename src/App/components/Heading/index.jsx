import P from 'prop-types';

function Heading({
  Head = 'h1',
  children,
  size,
  colorDark = false,
  uppercase = false,
  weight = 'font-600',
}) {
  return (
    <Head
      className={`${weight} ${colorDark ? 'text-white' : 'text-primary'}
         ${uppercase && `uppercase`} font-serrat
      ${size}  ${size == `text-large` && `md:text-large`} ${size == `text-large` && `text-mediumlarge`} leading-none`}
    >
      {children}
    </Head>
  );
}
Heading.propTypes = {
  Head: P.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']).isRequired,
  children: P.string.isRequired,

  size: P.string.isRequired,
  colorDark: P.bool,
  uppercase: P.bool,
  weight: P.string,
};
export default Heading;
