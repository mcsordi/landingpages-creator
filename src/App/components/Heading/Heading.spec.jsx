import { render, screen } from '@testing-library/react';
import Heading from '.';

describe('<Heading/>', () => {
  it('should render a heading on the screen with default properties', () => {
    render(<Heading size="text-large">hello</Heading>);
    const heading = screen.getByRole('heading', { name: 'hello' });

    expect(heading).toHaveClass('font-serrat');

    expect(heading).toHaveClass('text-primary');
    expect(heading).toHaveClass('text-mediumlarge');
    expect(heading).toHaveClass('leading-none');
    expect(heading).toHaveClass('md:text-large');
  });
  it('should render an uppercase heading', () => {
    render(
      <Heading uppercase size="text-large">
        hello
      </Heading>,
    );
    const heading = screen.getByRole('heading', { name: 'hello' });
    expect(heading).toHaveClass('uppercase');
  });
  it('should render a white heading', () => {
    render(
      <Heading colorDark size="text-large">
        hello
      </Heading>,
    );
    const heading = screen.getByRole('heading', { name: 'hello' });
    expect(heading).toHaveClass('text-white');
  });

  it('should render a md:text-large and a text-mediumlarge', () => {
    render(<Heading size="text-large">hello</Heading>);
    const heading = screen.getByRole('heading', { name: 'hello' });
    expect(heading).toHaveClass('text-large');
    expect(heading).toHaveClass('md:text-large');
    expect(heading).toHaveClass('text-mediumlarge');
  });
  it('should render a correct heading', () => {
    const { container } = render(
      <Heading Head="h5" size="text-large">
        hello
      </Heading>,
    );
    const h5 = container.querySelector('h5');

    expect(h5.tagName.toLowerCase()).toBe('h5');
  });
  it('should match a snapshot', () => {
    const { container } = render(
      <Heading colorDark uppercase Head="h5" size="text-large">
        hello
      </Heading>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
