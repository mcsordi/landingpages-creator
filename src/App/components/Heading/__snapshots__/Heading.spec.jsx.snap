// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading/> should match a snapshot 1`] = `
<h5
  class="font-600 text-white
         uppercase font-serrat
      text-large  md:text-large text-mediumlarge leading-none"
>
  hello
</h5>
`;
